### ISSUE TITLE
startNakadi not working (docker compose plus docker for mac)

### ISSUE THREAD

**Issue by dehora** [zalando/nakadi#468]
From https://github.com/zalando/nakadi/issues/439#issuecomment-248659201

> I've been having issues running Nakadi locally for some time now as well when using ./gradlew startNakadi. 

I've had a similar experience since the project switched over to compose with v2 syntax.

From https://github.com/zalando/nakadi/pull/414#issuecomment-250667280 it seems this combination is known to be an issue:

> docker for mac has a bug docker/docker#22753 and it event does not work for the current docker-compose file.

I understand the benefits to compose, but it's not ideal to exclude developers like this, especially those working on clients to support the project. Ideally it needs to be documented how to get up and running or that there are known problems with the project's current choice of docker configuration. In the meantime this can act as a covering ticket until it's fixed.

/ @adyach @dr4ke616 @kanuku @rcillo @valgog

**Comment by adyach** [comment]
@dehora as far as I understand the purpose of docker-compose file was to address development process of Nakadi itself and unload gradle build file. I always thought that for the integration purposes it is better to use integration environment like sandbox or staging which is fully set up to the latest available version.

Also we mentioned in our README file how to start Nakadi especially for Mac OS:
[this](https://github.com/zalando/nakadi#mac-os-docker-settings) and [this](https://github.com/zalando/nakadi#dependencies) 

Nevertheless, I strongly believe we have to support our users to simplify integrations with Nakadi.

**Comment by dehora** [comment]
Ok, I've made some progress:

- remove docker for mac (1.12.1)
- install docker toolbox

and with that setup, `docker-compose up -d --build` will bring up a cluster.

<img width="1214" alt="screen shot 2016-10-18 at 21 24 34" src="https://cloud.githubusercontent.com/assets/5454/19494972/5c36bbb0-9579-11e6-9305-21f64cc551d8.png">

The cluster functions within docker, but the 8080 port isn't forwarded to the host mac. That requires port forwarding via virtualbox:

![pf](https://cloud.githubusercontent.com/assets/5454/19495111/e9a5644c-9579-11e6-81a9-b513ae060837.gif)

and then `startNakadi` will stop spinning since the server is available:

![pf1](https://cloud.githubusercontent.com/assets/5454/19495318/cdbcb46e-957a-11e6-9ca0-49b2bfed086b.gif)


A couple of things then:

If the project is not going to support anyone using docker for mac, then readme needs to be updated to provide clear instructions. Macs are super-common in Zalando (and in general), and the docker project leans towards telling people to use docker for mac rather than [docker toolbox](https://github.com/docker/toolbox/releases/tag/v1.12.2): 

![pf2](https://cloud.githubusercontent.com/assets/5454/19495427/3be5d8a8-957b-11e6-9983-b7b4e571d6ac.gif)



As an aside confluent's docker repo has this warning as a headline item in its [readme file](https://github.com/confluentinc/cp-docker-images/blob/master/README.md) and I think Nakadi needs something similar. 

I get that most people on this ticket will figure out the workaround, but I'm thinking of the next 100 developers that will end up banging their head against this.

The alternative is running all four susbsytems in a single container. I know that's not idiomatic docker, and the project has deliberately moved anyway from it to single function containers, but the majority of people working with Nakadi just need something that comes up with minimal fuss. Either way, the documentation badly needs a update.

**Comment by dehora** [comment]
@adyach @dr4ke616 @kanuku @rcillo @valgog I guess in conclusion then, a doc update and some animated howto gifs should be enough to fix this for mac users.

**Comment by valgog** [comment]
Does not portmapping work?

**Comment by valgog** [comment]
@dehora I think it can still work in Docker for Mac

See `docker-compose run --service-ports`

Also https://docs.docker.com/compose/faq/#/whats-the-difference-between-up-run-and-start

It looks like one should not use `docker-compose up` for testing things. 

Maybe would should put `gradle` into a container and run it from `docker-compose` with dependencies :D

**Comment by valgog** [comment]
@v-stepanov Slava, do you remember why we needed to pin the network type to `host` in `docker-container`?

**Comment by dehora** [comment]
> Does not portmapping work?

In Docker for Mac, not via Kismatic's manual settings at least; port settings were not being preserved through the UI. Kismatic with docker toolbox has the same behaviour:

![pf3](https://cloud.githubusercontent.com/assets/5454/19520657/e1c7e76a-9608-11e6-9a4f-1a796dd21f9a.gif)

Ii'm guessing you'd can't edit them after the fact with this kind of compose setup - I hadn't seen this in Kismatic until yesterday. But editing port mappings via virtualbox works.

 Also iirc, the nodes weren't finding each other using d4m never mind being accessible from the host machine (I'd need to reinstall d4m to confirm that).

**Comment by valgog** [comment]
@dehora it looks like there is not way to force `docker-compose` to perform `up` command and also map the ports :(

Another issue is that `network_type: host` seems to be not working in Docker for Mac

**Comment by cesarcneto** [comment]
Just wanted to emphasize @dehora 's issue. 

I have just run through all these issues mentioned here. Mainly because I followed README without checking issues -- yes, shame on me :) -- and, in the end, I just removed `network_type:`  declaration from `docker-compose.yml` (which defaults to `bridge`) and used `docker-compose run --service-ports` as @valgog also suggested.

Since Nakadi is already open sourced, maybe you guys could think of having a `docker-compose-dev.yml` file to make the development process easier, like this [docker-compose.txt](https://github.com/zalando/nakadi/files/677240/docker-compose.txt), and update the README as well :)
